vaultclient/src/main/java/com/scality/vaultclient/dto/ListAccountsRequestMarshaller.java

```diff
@@ -0,0 +1,78 @@
+package com.scality.vaultclient.dto;
+
+import com.amazonaws.protocol.MarshallLocation;
+import com.amazonaws.protocol.MarshallingInfo;
+import com.amazonaws.protocol.MarshallingType;
+import com.amazonaws.protocol.ProtocolRequestMarshaller;
+import com.amazonaws.util.StringUtils;
+import com.scality.vaultclient.services.VaultClientException;
+import com.scality.vaultclient.utils.VaultClientUtils;
+
+/**
+ * The type List accounts request marshaller extends GenericRequestMarshaller to marshall List Account Requests.
+ */
+public class ListAccountsRequestMarshaller extends GenericRequestMarshaller<ListAccountsRequestDTO>{
+
+    private static final MarshallingInfo<String> MARKER_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("Marker").build();
+    private static final MarshallingInfo<Integer> MAX_ITEMS_BINDING = MarshallingInfo.builder(MarshallingType.INTEGER).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("MaxItems").build();
+    private static final MarshallingInfo<String> FILTER_KEY_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("filterKey").build();
+    private static final MarshallingInfo<String> FILTER_KEY_STARTS_WITH_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("filterKeyStartsWith").build();
+
+
+    private static final String LIST_ACCOUNTS_ACTION = "ListAccounts";
+
+    private static final ListAccountsRequestMarshaller instance = new ListAccountsRequestMarshaller();
+
+    public static ListAccountsRequestMarshaller getInstance() {
+        return instance;
+    }
+
+    /**
+     * Binds protocolMarshaller with listAccountsRequestDTO parameters to marshall.
+     *
+     * @param listAccountsRequestDTO         the request dto of type ListAccountsRequestDTO
+     * @param protocolMarshaller the protocol marshaller
+     */
+    @Override
+    public void marshall(ListAccountsRequestDTO listAccountsRequestDTO, ProtocolRequestMarshaller<ListAccountsRequestDTO> protocolMarshaller) {
+
+        super.marshall(listAccountsRequestDTO, protocolMarshaller, LIST_ACCOUNTS_ACTION);
+
+        try {
+            if (listAccountsRequestDTO.getMaxItems() < 0 || listAccountsRequestDTO.getMaxItems() > 1000) {
+                throw new VaultClientException("maxItems need to be a value between 1 and 1000 included");
```

Comment:
Shouldn't the `> 1000` case be handled server-side by returning an "invalid request" response? Hardcoding this in the client means that a new client would need to be released whenever Vault changes this limit up or down.

The `< 0` case is obviously invalid and should be handled client-side and there's little risk of it changing server-side.

Reply:
@RachedBenMustapha  I tried to emulate same behavior as `vaultclient`.
Ref: https://github.com/scality/vaultclient/blob/59356bf133f20d9ea0a74db64e27dd803952fb54/lib/IAMClient.js#L352

Reply:
Fixed after slack discussion to not have server side eligible validations on client.
Fix commit b225694